### PR TITLE
[fix] Mount bind readonly not really readonly

### PR DESCRIPTION
**Pull request by zamentur**
Untested
Fix https://dev.yunohost.org/issues/991

### CODE REVIEW

File: src/yunohost/backup.py [hunk 1]
```diff
@@ -1543,23 +1544,37 @@ def _organize_files(self):
...
+
+                try:
+                    run_commands(mount_commands)
+                except Exception as e:
+                    logger.warning(m18n.n("backup_couldnt_bind", src=src, dest=dest))
+                    # To check if dest is mounted, use /proc/mounts that
+                    # escape spaces as \040
+                    is_mounted = "cat /proc/mounts | cut -d ' ' -f 2 | sed -s 's/\\\\040/ /g' | grep -xq '%s'" % dest
+                    if os.system(is_mounted) == 0:
+			umount_command = "umount -R %s" % dest
```
Comment (zamentur) [inline src/yunohost/backup.py]: Idem

File: src/yunohost/backup.py [hunk 2]
```diff
@@ -1543,23 +1544,37 @@ def _organize_files(self):
...
-                ret = subprocess.call(["mount", "-r", "--rbind", src, dest])
-                if ret == 0:
-                    continue
+
+                mount_commands = []
+                mount_commands.append("mount --rbind %s %s" % (src, dest))
+                mount_commands.append("mount -o remount,ro,bind %s" % dest)
+
+                try:
+                    run_commands(mount_commands)
```
Comment (zamentur) [inline src/yunohost/backup.py]: I think subprocess.call is better, how runcommand protects from a dest arg with malicious code ?

File: src/yunohost/backup.py [hunk 3]
```diff
@@ -1543,23 +1544,37 @@ def _organize_files(self):
...
                 else:
-                    logger.warning(m18n.n("bind_mouting_disable"))
-                    subprocess.call(["mountpoint", "-q", dest,
-                                    "&&", "umount", "-R", dest])
+                    # Success, go to next file to organize
+                    continue
+
+            # For files, create a hardlink
             elif os.path.isfile(src) or os.path.islink(src):
-                # Create a hardlink if src and dest are on the filesystem
```
Comment (zamentur) [inline src/yunohost/backup.py]: The goal was to explain you cannot create a hardlink on different fs

Comment (alexAubin) [inline src/yunohost/backup.py]: Sorry 😕 :P  I put a comment back in 36f256e

**Comment by alexAubin** [comment]
(@zamentur : you okay with the changes ?)

**Comment by zamentur** [comment]
Yes i agree with this change

**Pull request merged**